### PR TITLE
fix Telekinesis teleporting some items

### LINKED ISSUE #55885: Telekinesis teleports black boxes and body bags to you
## Round ID:
None, found the issue noted on another server, loaded up my tgrepo to test for myself and confirmed its a thing

## Reproduction:

basically, telekensis teleports the blackbox into your hand, the bluespace body bag (didnt test normal BBs though they should function the same, only that a normal one wouldnt be closed if it had items in it), MMI (though couldnt actually test it personally due to one connection on my host). and perhaps other things as well i didnt test for

I assume the issue is simply that these are items that when attacked by hand spawn the item into your hand, but since telekensis isnt an actual hand itself, the next best thing occurs and it enters your hand directly (or at your feet in the case of the bodybag)

not sure how easy this would be to fix, though the simplest would be to just disable TK from activating things that cause items to spawn into your hand, im also not sure if this is a "bug" or a "feature" since well, it does add one way of stealing the blackbox, but on the other hand, doesnt look like its intended since well, its *too easy*, it would make more sense if activating the black box machine with TK that it spawns the black box on the machine, and you gotta find a way to TK it out or something, or using TK on the body bag SHOULD open on the tile its on (it opens and closes it fine IF the bodybags already unfolded, its only when folded that unfolding it teleports it)

((i checked the issues section for any issues on this particular issue and couldnt find any, apologies if this issue is somehow already made before, I did however see a similar issue [#52882] with teleporting beakers which was listed as a bug as well, hence why i assume this one is too))

merging #55916 will fix the issue

https://user-images.githubusercontent.com/40489693/103445118-32450700-4c3e-11eb-8177-cfe88ad14c77.mp4

### PR DESCRIPTION
**Pull request by AlinaStarkova**
Fixed telekinesis from teleporting the Blackbox, bodybags and all its contents, and brains from MMIs - Nari

## About The Pull Request

as outlined in issue #55885, telekinesis was teleporting stuff like the blackbox, brains from an MMI, bodybags (including all the contents if bluespace) when activated with a telekinetic hand

fixed by simply having them drop onto the tile its on when using telekinesis, rather than enter the users hand

(the conflicts is caused by a broken PR that got merged, its not to do with my code so e)
https://user-images.githubusercontent.com/40489693/103470740-f85e2880-4d43-11eb-8f19-112a09b4d0a2.mp4

## Why It's Good For The Game

as neato as it is for a way to teleport your objective right to you, among other things, this is a bug with the way TK functions, and TK is just not feasible to overhaul for this

fixes #55885

## Changelog
:cl: Nari Harimoto
fix: Telekinesis no longer teleports the Blackbox, brains from an MMI, or bodybags with all its contents, directly to you
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
Not all items right? just some of em

**Comment by AlinaStarkova** [comment]
> 
> 
> Not all items right? just some of em

its just those 3, e, it would be a right pain to check every single item in the game to make them not put items into your hand when activated

File: code/game/machinery/telecomms/machines/message_server.dm [hunk 1]
```diff
@@ -24,7 +24,7 @@
 /obj/machinery/blackbox_recorder/attack_hand(mob/living/user)
 	. = ..()
 	if(stored)
-		user.put_in_hands(stored)
+		stored.forceMove(get_turf(src))
```
Comment (Jared-Fogle) [inline code/game/machinery/telecomms/machines/message_server.dm:27]: Use `drop_location()` instead, here and everywhere else.

**Comment by Jared-Fogle** [comment]
>  (does make it less convenient, however this was the fix utilized for stopping AIs from ejecting beakers from machines spawning them onto the AI core)

I think you should just check if they're adjacent after dropping it, and put it into their hands then.

File: code/game/machinery/telecomms/machines/message_server.dm (hunk 1 above)
Comment (AlinaStarkova) [reply code/game/machinery/telecomms/machines/message_server.dm:27]: i tried, i really did, i tried drop_location(stored) as well as stored.drop_location() .... all those did, was *delete* the items they were supposed to drop, like, completely vanish. force moving it however worked flawlessly

**Comment by AlinaStarkova** [comment]
> 
> 
> > (does make it less convenient, however this was the fix utilized for stopping AIs from ejecting beakers from machines spawning them onto the AI core)
> 
> I think you should just check if they're adjacent after dropping it, and put it into their hands then.

hmmm, let me go code browsing to find out how to do this

**Comment by Jared-Fogle** [comment]
```dm
if (thing_being_dropped.Adjacent(user))
    user.put_in_hands(thing_being_dropped)
```

File: code/game/machinery/telecomms/machines/message_server.dm (hunk 1 above)
Comment (MrDoomBringer) [reply code/game/machinery/telecomms/machines/message_server.dm]: I believe @Jared-Fogle meant `stored.forceMove(drop_location())`
If you're ever unsure of how to use a particular function or whatever, one way to quickly figure out is to search the codebase for example use cases! :)
image

If you search for the proc definition (or ctrl+click a function call if you're using vscode and the .dm extensions) you can sometimes see helpful comments explaining the function further.
image

Comment (AlinaStarkova) [reply code/game/machinery/telecomms/machines/message_server.dm]: this is outdated, its already added in.

**Comment by Jared-Fogle** [comment]
Please merge from latest upstream, there was a CI failure on master that has since been fixed.

File: code/game/objects/items/bodybag.dm [hunk 2]
```diff
@@ -8,7 +8,10 @@
 	var/unfoldedbag_path = /obj/structure/closet/body_bag
 
 /obj/item/bodybag/attack_self(mob/user)
-	deploy_bodybag(user, user.loc)
+	if(user.is_holding(src))
+		deploy_bodybag(user, user.loc)
+	else
+		deploy_bodybag(user, src.loc)
```
Comment (Jared-Fogle) [inline code/game/objects/items/bodybag.dm:14]:
```suggestion
		deploy_bodybag(user, get_turf(src))
```

File: code/game/objects/items/bodybag.dm [hunk 3]
```diff
@@ -8,7 +8,10 @@
 	var/unfoldedbag_path = /obj/structure/closet/body_bag
 
 /obj/item/bodybag/attack_self(mob/user)
-	deploy_bodybag(user, user.loc)
+	if(user.is_holding(src))
+		deploy_bodybag(user, user.loc)
```
Comment (Jared-Fogle) [inline code/game/objects/items/bodybag.dm:12]:
```suggestion
		deploy_bodybag(user, get_turf(user))
```

**Comment by ArcaneMusic** [comment]
If you change the wording on your PR body to something along the lines of "Fixes #issuenumber" it will close the issue once merged, btw!

**Comment by AlinaStarkova** [comment]
> If you change the wording on your PR body to something along the lines of "Fixes #issuenumber" it will close the issue once merged, btw!

huh, i added the issue number in and it didnt link the issues, even though another similar PR i did, did link. i was wondering why it didnt link this time

File: code/game/objects/items/bodybag.dm (hunk 2 above)
Comment (AlinaStarkova) [reply code/game/objects/items/bodybag.dm]: done

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is now above zero (0). Feel free to make Feature/Tweak/Balance PRs.

**Pull request merged**